react-components/src/components/AccountDetails/useAccountDetails.js

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+import { useMemo, useCallback, useState } from 'react';
+import { useQuery, useMutation } from '@apollo/react-hooks';
+import { useUserContext } from '../../context/UserContext';
+
+const useAccountDetails = props => {
```

Comment:
Since this hook stores state, I think it would be appropriate to use a React context here. So you don't need to pass all the exported logic and state here via props to child components like `EditModal` and `EditForm`.

Reply:
There are only a few components under this tree, I don't see any reason to use a React Context just to manage the state of a modal. In general, I don't see any problem with passing state down to child components for trivial matters. I see React Contexts for cross-cutting concerns, like make some "context data" available to components that are not part of the same sub-tree.